### PR TITLE
Build framework for iOS

### PR DESCRIPTION
**Pull request by mattmassicotte**
Here's an interesting one. I was playing around with building this for iOS and the simulator. It was a pain, because there was so much duplication in the script. So, I used my Super Bash Skills to make a function! It is true that this makes some assumptions about the parsers using the same flags. But, some still need special handling (like checking on PRs...), so I don't think it's too bad to customize if/when this actually comes up.

Joking aside, I'm really not very good at shell scripting in general, so if there's anything you don't like or think could better, I'm all for it.

### CODE REVIEW

File: scripts/build-tree-sitter.sh [hunk 1]
```diff
@@ -4,98 +4,119 @@ set -euxo pipefail
...
 
-PKG_CONFIG_PATH="$TMP_BUILD_DIR/build/lib/pkgconfig"
-export PKG_CONFIG_PATH=${PKG_CONFIG_PATH:+:}${PKG_CONFIG_PATH}:$(pkg-config --variable pc_path pkg-config)
+IPHONEOS_SYSROOT=$(xcrun --sdk iphoneos --show-sdk-path)
+IPHONESIMULATOR_SYSROOT=$(xcrun --sdk iphonesimulator --show-sdk-path)
 
-COMMON_FLAGS="-arch arm64 -arch x86_64 -mmacosx-version-min=10.13"
-LANGUAGE_DATA_DIR="$SCRIPT_DIR/../Sources/tree_sitter_language_resources/LanguageResources"
+MACOS_COMMON_FLAGS="-arch arm64 -arch x86_64 -mmacosx-version-min=10.13"
+IPHONEOS_COMMON_FLAGS="-arch arm64 -miphoneos-version-min=11.0 -fembed-bitcode -isysroot $IPHONEOS_SYSROOT"
```
Comment (krzyzanowskim) [inline scripts/build-tree-sitter.sh:16]: I think the iOS slice suppose to go as low as `armv7s`?

Comment (mattmassicotte) [reply scripts/build-tree-sitter.sh:16]: I thought so too, but according to some research, it looks like its `arm64`-only as of iOS 11!

File: Sources/tree_sitter_language_resources/LanguageResources/go/structure.scm [hunk 2]
```diff
@@ -0,0 +1,175 @@
+(import_declaration
```
Comment (krzyzanowskim) [inline Sources/tree_sitter_language_resources/LanguageResources/go/structure.scm:1]: what are these files for?

Comment (mattmassicotte) [reply Sources/tree_sitter_language_resources/LanguageResources/go/structure.scm:1]: I should have explained originally.

But, it was me that added them to go and gomod. They are used to get information about the syntactic structure of the code, which I use for indentation calculations. They are also handy for finding matching elements, like `if`-`{`-`}`. They aren't *technically* part of this change, but got pulled in as part of rebuilding the framework from the parser sources.

Comment (krzyzanowskim) [reply Sources/tree_sitter_language_resources/LanguageResources/go/structure.scm:1]: thanks. nice!

**Pull request merged**